engines/director/castmember/sound.cpp

```diff
@@ -100,6 +100,16 @@ void SoundCastMember::load() {
 			// In older versions, always loop sounds that contain a loop start and end.
 			_looping = audio->hasLoopBounds();
 		}
+		else if (_cast->_version == kFileVer400) {
```

Comment:
This does not conform to our https://wiki.scummvm.org/index.php?title=Code_Formatting_Conventions.

Also, I would do it a bit differently:

```C++
if (!audio->hasValidLoopBounds()) {
   if (_cast->_version == kFileVer400) {
      debugC(2, kDebugLoading, "Cor....
      ...
    } else {
      warning("SoundCastMember::load(): invalid loop bounds detected");
    }
}
```
e.g. the idea is to catch the same situation if it happens in other Director versions, so we could test with the original and see what it does. Perhaps, the condition is `>= kFileVer400`.

If you have possibility to have this movie loaded in the later versions of Director, you could figure it out now.

Reply:
Yeah - I 'll rework it how you suggest

> e.g. the idea is to catch the same situation if it happens in other Director versions, so we could test with the original and see what it does. Perhaps, the condition is `>= kFileVer400`.

Importing into Windows D4, then re-saving fixes invalid loop bounds :
+  sets loop flag to 16 (as in disabling looping)
+  Adds valid loop bounds (0->sample length) into the sound resources

It feels to me like it's a migration away from using the sound resource loop bounds to determine looping. Maybe even stop using the soundmanager format for sound resources in later Director Versions ?  

I looked through the D4 demos also and found that out of the 200 or so demos and ~11,000 embedded sound cast members, every sound cast member in `kFileVer404` movies had valid loop points (`start < end <= length`). This suggests to me the issue does not need to be corrected at `kFileVer404`. As such, would it be better to leave it as the existing warning for now?

Reply:
Numbers from my demo scan here:  https://gist.github.com/threefins/fbc02fb1d184a8e06c0bd27bd4d29b07